src/Microsoft.DotNet.Darc/src/DarcLib.AzDev/AzureDevOpsClient.cs

```diff
@@ -1001,5 +994,35 @@ public async Task<AzureDevOpsReleaseDefinition> GetReleaseDefinitionAsync(string
 
             return content.ToObject<AzureDevOpsReleaseDefinition>();
         }
+
+        /// <summary>
+        // If repoUri includes the user in the account we remove it from URIs like
+        // https://dnceng@dev.azure.com/dnceng/internal/_git/repo
+        // If the URL host is of the form "dnceng.visualstudio.com" like
+        // https://dnceng.visualstudio.com/internal/_git/repo we replace it to "dev.azure.com/dnceng"
+        // for consistency
+        /// </summary>
+        /// <param name="url">The original url</param>
+        /// <returns>Transformed url</returns>
+        public static string NormalizeUrl(string repoUri)
```

Comment:
Can you add a few tests for this to ensure that this works properly (and doesn't replace github urls, etc.,)